src/Groups/types.jl

```diff
@@ -197,6 +229,53 @@ as shown in Section [Elements of groups](@ref elements_of_groups).
 """
 const PcGroupElem = BasicGAPGroupElem{PcGroup}
 
+
+"""
+    SubPcGroup
+
+Subgroup of a polycyclic group,
+a group that is defined by generators that are elements of a group `G`
+of type [`PcGroup`](@ref).
+The arithmetic operations with elements are thus performed using the
+polycyclic presentation of `G`.
+
+Operations for computing subgroups of a group of type `PcGroup` or
+`SubPcGroup`, such as `derived_subgroup` and `sylow_subgroup`,
+return groups of type `SubPcGroup`.
+"""
+@attributes mutable struct SubPcGroup <: GAPGroup
+  X::GapObj
+  full_group::PcGroup
+#T no, create embedding!
+
+  function SubPcGroup(G::GapObj)
+    @assert GAPWrap.IsPcGroup(G) || GAP.Globals.IsPcpGroup(G)
+    full = GAP.Globals.GroupOfPcgs(GAP.Globals.FamilyPcgs(G))
+#T use GAPWrap!
+    z = new(G, PcGroup(full))
+    return z
+  end
+end
+
+sub_pc_group(G::GapObj) = SubPcGroup(G)
+
+
+"""
+    SubPcGroupElem
+
+Element of a subgroup of a polycyclic group.
+
+# Examples
+
+```jldoctest
+
+... hier! ...
+
+```
+"""
+const SubPcGroupElem = BasicGAPGroupElem{SubPcGroup}
```

Comment:
To me the most annoying part of this PR actually seems to be `SubPcGroupElem`; we probably would want to allow multiplying a `SubPcGroupElem` by a `PcGroupElem` (and vice-versa), and possibly in various other contexts one may have to / wish to support both?

Perhaps a suitable `promote_rule` can help a bit?